### DISCUSSION TITLE
Probably bug in macos-docker integration, not julia, right?

### DISCUSSION THREAD

**Discussion by hbe72** [JuliaLang/julia#53315] [General]
I have tested host system is macos running on M2 and M3.
Docker Desktop 4.27.2 with having Docker engine 25.0.3
Ubuntu 22.04 container base image and installation of Julia 1.10 (tried also 1.9.4 and 1.8.5) from tar.gz. See attached Dockerfile.

Attempting a multiplatform build with docker buildx. The native part of multiplatform build succeeds, but the cross-compiled target fails. Thus I'm showing only the cross-compilation part of the multiplatform build below.

Thus first setup buildx
``
% docker buildx create --use --platform=linux/arm64,linux/amd64
``

Then the actual build
``
docker buildx build  --ssh default -t <your_registry>/<owner>/minimalfail:latest . --build-arg OWNER=<your_registry>/<owner> --platform "linux/amd64"
`` 

Container build works fine until julia is used within the container, and then the error is
``
0.071 Message:/opt/julia-1.10.0/bin/../lib/julia/libopenlibm.so: cannot enable executable stack as shared object requires: Invalid argument
`` 

Another way around, thus multiplatform build on macos running on X86 (thus the amd64 is the native and arm64 is the cross-target) everything works.

Find the minimal docker file attached:
[Dockerfile.tar.gz](https://github.com/JuliaLang/julia/files/14268846/Dockerfile.tar.gz)

**Comment by vtjnash** [comment]
Seems like an libopenlibm bug, though it was theoretically fixed a long time ago: https://github.com/JuliaMath/openlibm/pull/34. Perhaps a file got missed? More details about it here: https://wiki.gentoo.org/wiki/Hardened/GNU_stack_quickstart
